tests/topotests/lib/topojson.py

```diff
@@ -99,36 +100,35 @@ def build_topo_from_json(tgen, topo):
     while listRouters != []:
         curRouter = listRouters.pop(0)
         # Physical Interfaces
-        if "links" in topo["routers"][curRouter]:
-            for destRouterLink, data in sorted(
-                topo["routers"][curRouter]["links"].iteritems()
-            ):
-                currRouter_lo_json = topo["routers"][curRouter]["links"][destRouterLink]
+        if 'links' in topo['routers'][curRouter]:
+            for destRouterLink, data in sorted(topo['routers'][curRouter]['links']. \
+                                               iteritems()):
+                currRouter_lo_json = \
+                    topo['routers'][curRouter]['links'][destRouterLink]
                 # Loopback interfaces
-                if "type" in data and data["type"] == "loopback":
+                if 'type' in data and data['type'] == 'loopback':
                     router_count += 1
-                    if (
-                        "ipv4" in currRouter_lo_json
-                        and currRouter_lo_json["ipv4"] == "auto"
-                    ):
-                        currRouter_lo_json["ipv4"] = "{}{}.{}/{}".format(
-                            topo["lo_prefix"]["ipv4"],
-                            router_count,
-                            number_to_column(curRouter),
-                            topo["lo_prefix"]["v4mask"],
-                        )
-                    if (
-                        "ipv6" in currRouter_lo_json
-                        and currRouter_lo_json["ipv6"] == "auto"
-                    ):
-                        currRouter_lo_json["ipv6"] = "{}{}:{}/{}".format(
-                            topo["lo_prefix"]["ipv6"],
-                            router_count,
-                            number_to_column(curRouter),
-                            topo["lo_prefix"]["v6mask"],
-                        )
-
-                if "-" in destRouterLink:
+                    currRouter_lo_json['interface'] = destRouterLink
```

Comment:
Haven't we gone over this before?  Why are we mixing whitespace changes with other code changes.  It makes it extremely hard to review and is plain annoying to wade through.  Please take some time and go through your commits and break them *up*

Reply:
It is resolved, changes are merged via different PR.